__tests__/app.test.js

```diff
@@ -124,6 +124,102 @@ describe("Articles", () => {
+          expect(article[0].topic).toEqual(expect.any(String));
+          expect(article[0].author).toBe("rogersop");
+          expect(article[0].body).toEqual(expect.any(String));
+          expect(article[0].created_at).toEqual(expect.any(String));
+          expect(article[0].votes).toEqual(expect.any(Number));
+          expect(article[0].comment_count).toEqual(expect.any(Number));
+        });
+    });
+    test("status: 404 when given a non-existent sortby", () => {
+      return request(app)
+        .get("/api/articles?sort_by=writer")
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Query does not exist");
+        });
+    });
+    test("orderby orders by default desc but can also asc", () => {
+      return request(app)
+        .get("/api/articles?sort_by=title&order_by=ASC")
+        .expect(200)
+        .then(({ body: { article } }) => {
+          expect(article).toHaveLength(12);
+          expect(article[0].article_id).toBe(6);
+          expect(article[0].title).toBe("A");
+          expect(article[0].topic).toEqual(expect.any(String));
+          expect(article[0].author).toEqual(expect.any(String));
+          expect(article[0].body).toEqual(expect.any(String));
+          expect(article[0].created_at).toEqual(expect.any(String));
+          expect(article[0].votes).toEqual(expect.any(Number));
+          expect(article[0].comment_count).toEqual(expect.any(Number));
+        });
+    });
+    test("status: 404 when given a non-existent orderby", () => {
+      return request(app)
+        .get("/api/articles?order_by=up")
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Query does not exist");
+        });
+    });
+    test("topic filters articles by topic value in query", () => {
+      return request(app)
+        .get("/api/articles?topic=cats")
+        .expect(200)
+        .then(({ body: { article } }) => {
+          expect(article).toHaveLength(1);
+          expect(article[0].article_id).toEqual(expect.any(Number));
+          expect(article[0].title).toEqual(expect.any(String));
+          expect(article[0].topic).toBe("cats");
+          expect(article[0].author).toEqual(expect.any(String));
+          expect(article[0].body).toEqual(expect.any(String));
+          expect(article[0].created_at).toEqual(expect.any(String));
+          expect(article[0].votes).toEqual(expect.any(Number));
+          expect(article[0].comment_count).toEqual(expect.any(Number));
+        });
+    });
+    test("status: 404 when given a non-existent topic", () => {
+      return request(app)
+        .get("/api/articles?topic=none")
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Query does not exist");
+        });
+    });
+    test("status: 200 and default queries when given a non-existent topic", () => {
+      return request(app)
+        .get("/api/articles?none=none")
```

Comment:
This should be topic = none

Reply:
It was meant to be where the query is non-existent

Reply:
I have changed it in the new upload :)